core/app/models/spree/order_inventory.rb

```diff
@@ -53,12 +53,12 @@ def remove(item_units, shipment = nil)
     # first unshipped that already includes this variant
     # first unshipped that's leaving from a stock_location that stocks this variant
     def determine_target_shipment
-      shipment = order.shipments.detect do |shipment|
-        shipment.ready_or_pending? && shipment.include?(variant)
+      shipment = order.shipments.detect do |s|
+        s.ready_or_pending? && s.include?(variant)
       end
 
-      shipment ||= order.shipments.detect do |shipment|
-        shipment.ready_or_pending? && variant.stock_location_ids.include?(shipment.stock_location_id)
+      shipment || order.shipments.detect do |s|
+        s.ready_or_pending? && variant.stock_location_ids.include?(s.stock_location_id)
```

Comment:
I went with a different solution I hope is acceptable: dropped the outer variable and kept the `shipment` variable in the block.